### DISCUSSION TITLE
【CDC消费速率】flink 使用 mongo sql cdc-connector 如何提高 source 端消费速率？

### DISCUSSION THREAD

**Discussion by maben996** [apache/flink-cdc#2779] [Q&A]
UI上看到QPS在2万左右，并且source的并行度一直是1。请问如何提高source的并行度？以及并行度的提升是否有助于提升消费速率？
<img width="1434" alt="image" src="https://github.com/ververica/flink-cdc-connectors/assets/44827813/c601e3c9-3d4e-4bc3-8fae-a9ab40c649e1">
sql 的 source  table 配置如下，这里在[CDC文档](https://github.com/ververica/flink-cdc-connectors/blob/master/docs/content/connectors/mongodb-cdc(ZH).md)中没有找到可以配置source并行度的地方。提高了拉数据的batch size。发现对source的QPS没有提升效果。
<img width="1056" alt="image" src="https://github.com/ververica/flink-cdc-connectors/assets/44827813/f10dc421-9eff-4358-98ad-4082741e1a54">
flink.conf文件中的默认并行度配置如下
<img width="592" alt="image" src="https://github.com/ververica/flink-cdc-connectors/assets/44827813/362be998-71b7-4d4c-9073-c0f147d42c0a">
业务背景：有一批百亿级别的数据需要同步从mongo同步到doris，存量+增量都要同步，所以想使用CDC同步，但是发现速率太慢，目前QPS 2w，预计300亿要同步20天左右才能跑完。

**Comment by gtk96** [comment]
@maben996 2.3.0 及以上全量阶段是可以多并行度同步的，`SET 'parallelism.default' = 8;` 增量阶段目前只能单并行度

**Reply by maben996** [reply]
@gtk96 全量阶段是指配置为 scan.startup.mode = 'initial' 吗？

**Reply by gtk96** [reply]
是的

**Comment by maben996** [comment]
@gtk96 实测了一下，只有当'scan.incremental.snapshot.enabled' = 'true' 和 'scan.startup.mode' = 'initial' 都配置的时候，source回使用默认并行度，只开'scan.startup.mode' = 'initial'  source并行度还是1。
<img width="993" alt="image" src="https://github.com/ververica/flink-cdc-connectors/assets/44827813/a5f337bb-36ac-4adb-9b7c-865a923038d2">
但是当并行度为4时，source单个并行度的吞吐量 5k~6k* 4 也是2w多的消费速率，和之前source并行为1的消费速度相比几乎没有提升。
<img width="1631" alt="image" src="https://github.com/ververica/flink-cdc-connectors/assets/44827813/26848274-b05d-40ba-84e1-de1138751f81">
这种情况合理吗？source并行度增加但是消费速率不变

**Comment by gtk96** [comment]
> @gtk96 实测了一下，只有当'scan.incremental.snapshot.enabled' = 'true' 和 'scan.startup.mode' = 'initial' 都配置的时候，source回使用默认并行度，只开'scan.startup.mode' = 'initial' source并行度还是1。 <img alt="image" width="993" src="https://private-user-images.githubusercontent.com/44827813/287909385-a5f337bb-36ac-4adb-9b7c-865a923038d2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDE3Njg0MjAsIm5iZiI6MTcwMTc2ODEyMCwicGF0aCI6Ii80NDgyNzgxMy8yODc5MDkzODUtYTVmMzM3YmItMzZhYy00YWRiLTliN2MtODY1YTkyMzAzOGQyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjA1VDA5MjIwMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWJmYzQ1MjcwYjkyOTViOTU3ODBjZWRiOTc1OTNhZTIwZmZhMzBhOTNmZGViMGVkMTljOTA3MDc5MGU2MTBkMGImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.6Nrj-kZYX1dGLPNRFMQNhTjxN2bJAVfwBhZacDh4E6I"> 但是当并行度为4时，source单个并行度的吞吐量 5k~6k* 4 也是2w多的消费速率，和之前source并行为1的消费速度相比几乎没有提升。 <img alt="image" width="1631" src="https://private-user-images.githubusercontent.com/44827813/287909773-26848274-b05d-40ba-84e1-de1138751f81.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDE3Njg0MjAsIm5iZiI6MTcwMTc2ODEyMCwicGF0aCI6Ii80NDgyNzgxMy8yODc5MDk3NzMtMjY4NDgyNzQtYjA1ZC00MGJhLTg0ZTEtZGUxMTM4NzUxZjgxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjA1VDA5MjIwMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQ3YjJiMGUyZTY4NWEwYWE2YmU1NzM0ODVlYTdmOWE0OTlmOGVhZGYyZWY0ZGRkZWZmOTk5NmY0YWU2ZTk4NWEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.UWU83fyBTO2V1vxvVkmApH_EP06bNYJ7ylS5HUMN7wY"> 这种情况合理吗？source并行度增加但是消费速率不变

增加资源会不会快一点呢？

**Reply by maben996** [reply]
回答粘贴错了吧...哈哈

**Comment by gtk96** [comment]
> 回答粘贴错了吧...哈哈

理论上讲，source 端没有性能瓶颈的，瓶颈主要是 sink 端，所以你在 source 资源，包括拉取数据量都可以放大一些

**Comment by yangyuelong** [comment]
问题解决了么？
我也遇到了同样的问题，我设置了source并行度=10，web ui侧source端只有一个subtask接收到了数据。 这很诡异。
![Uploading image.png…]()

**Reply by night-create** [reply]
The default value for debezium.tasks.max is 1.The maximum number of tasks to create for this connector. Because the MySQL connector always uses a single task, changing the default value has no effect.
